crates/circuit/src/interner.rs

```diff
@@ -206,7 +206,8 @@ mod test {
     fn default_key_exists() {
         let mut interner = Interner::<[u32]>::new();
         assert_eq!(interner.get_default(), interner.get_default());
-        assert_eq!(interner.get(interner.get_default()), &[]);
+        let res: &[u32] = &[];
+        assert_eq!(interner.get(interner.get_default()), res);
```

Comment:
What's going on here?

Reply:
TBH I can't remember. It's either the compiler started complaining here and suggested I do this to make it work, or I was debugging something and did this to do `println!("{:?}", res)` before the assertion and forgot to change it back but removed the print. I'll try reverting this and see what happens.

Reply:
After trying it locally I remember now. The compiler was complaining it can't infer the type of `&[]` anymore so I used the res var to define it as a `&[u32]`.

Reply:
This seems weird that the PyO3 update would cause the inference here to suddenly fail.  The types of `Interner` and `get_default()` shouldn't have changed.